lib/datadog/core/configuration/settings.rb

```diff
@@ -646,23 +646,13 @@ def initialize(*_)
           # @default `nil`.
           # @return [Array<String>,nil]
           option :extra_services do |o|
+            o.type :array
+            o.default []
             o.setter do |v|
-              if v.nil?
-                []
-              elsif !v.is_a?(Array)
-                Datadog.logger.warn(
-                  "Unexpected remote.extra_services type #{v.class}, expected Array. " \
-                  'Using default value `[]`'
-                )
-
-                []
-              elsif (t = v.find { |e| !e.is_a?(String) })
-                Datadog.logger.warn(
-                  "Unexpected remote.extra_services element type #{t.class}, expected String. " \
-                  'Using default value `[]`'
-                )
+              if v.any? { |e| !e.is_a?(String) }
+                t = v.find { |e| !e.is_a?(String) }
```

Comment:
The previous code had an interesting corner case: with `[nil]` it would return `nil`, which would be falsey, thus pass the condition and not produce the expected error!

The double iteration is fine since it's on the unhappy path.